addons/mail/static/src/messaging/component/discuss_sidebar/discuss_sidebar.js

```diff
@@ -71,35 +96,91 @@ class DiscussSidebar extends Component {
 
     /**
      * @private
-     * @returns {Object}
+     */
+    _update() {
+        if (this._quickSearchInputRef.el) {
+            this._quickSearchInputRef.el.value = this.discuss.sidebarQuickSearchValue;
+        }
+    }
+
+    /**
+     * @private
+     * @return {Object}
      */
     _useStoreCompareDepth() {
         return {
-            pinnedChannelList: 1,
-            pinnedChatList: 1,
-            pinnedMailboxList: 1,
-            pinnedMailChannelAmount: 1,
+            allOrderedAndPinnedChats: 1,
+            allOrderedAndPinnedMailboxes: 1,
+            allOrderedAndPinnedMultiUserChannels: 1,
+            allPinnedChannelAmount: 1,
         };
     }
+
     /**
      * @private
-     * @param {Object} state
      * @param {Object} props
-     * @returns {Object}
+     * @return {Object}
      */
-    _useStoreSelector(state, props) {
+    _useStoreSelector(props) {
         return {
-            pinnedChannelList: this.storeGetters.pinnedChannelList(),
-            pinnedChatList: this.storeGetters.pinnedChatList(),
-            pinnedMailboxList: this.storeGetters.pinnedMailboxList(),
-            pinnedMailChannelAmount: this.storeGetters.pinnedMailChannelAmount(),
+            allOrderedAndPinnedChats:
+                this.env.entities.Thread.allOrderedAndPinnedChats,
```

Comment:
in this case I would extract Thread to a variable because too much repetition/new line